shotover-proxy/tests/runner/hotreload_int_tests.rs

```diff
@@ -24,38 +26,74 @@ async fn test_hotreload_basic_valkey_connection() {
 
-    //Second shotover
-    let shotover_b = shotover_process("tests/test-configs/hotreload/topology-alt.yaml")
-        .with_hotreload_from_socket(socket_path)
+    // Establish connection to old instance and store test data
+    let client_old = Client::open("valkey://127.0.0.1:6380").unwrap();
+    let mut con_old = client_old.get_connection().unwrap();
+
+    // Store data in the old instance
+    let _: () = con_old
+        .set("persistent_key", "data_from_old_instance")
+        .unwrap();
+    let _: () = con_old.set("counter", 1).unwrap();
+
+    // Verify data is accessible through old instance
+    let stored_value: String = con_old.get("persistent_key").unwrap();
+    assert_eq!(stored_value, "data_from_old_instance");
+
+    // Now start the new instance that will request hot reload
+    let shotover_new = shotover_process("tests/test-configs/hotreload/topology.yaml")
+        .with_hotreload(true)
         .with_log_name("shot_new")
+        .with_hotreload_socket_path("/tmp/shotover-new.sock") // Different socket for new instance
+        .with_hotreload_from_socket(socket_path) // Request handoff from old instance
         .with_config("tests/test-configs/shotover-config/config_metrics_disabled.yaml")
         .start()
         .await;
-    let client_b = Client::open("valkey://127.0.0.1:6381").unwrap();
-    let mut con_b = client_b.get_connection().unwrap();
-    let _: () = con_b.set("key_from_b", "value_from_b").unwrap();
 
-    let value_a_from_b: String = con_b.get("key_from_a").unwrap();
-    assert_eq!(value_a_from_b, "value_from_a");
-    let value_b_from_a: String = con_a.get("key_from_b").unwrap();
-    assert_eq!(value_b_from_a, "value_from_b");
+    let client_new = Client::open("valkey://127.0.0.1:6380").unwrap();
+    let mut con_new = client_new
+        .get_connection()
+        .expect("Failed to connect to new instance");
+
+    // Verify that data stored in old instance is still accessible
+    let persistent_value: String = con_new.get("persistent_key").unwrap();
+    assert_eq!(
+        persistent_value, "data_from_old_instance",
+        "Data should persist after hot reload socket handoff"
+    );
+
+    // Verify new instance can handle new operations
+    let _: () = con_new.set("new_key", "data_from_new_instance").unwrap();
+    let new_value: String = con_new.get("new_key").unwrap();
+    assert_eq!(new_value, "data_from_new_instance");
+
+    // Test that we can increment the counter
+    let counter_value: i32 = con_new.incr("counter", 1).unwrap();
+    assert_eq!(
+        counter_value, 2,
+        "Counter should increment from value set by old instance"
+    );
+
+    // Verify ping works
+    let pong: String = redis::cmd("PING").query(&mut con_new).unwrap();
+    assert_eq!(pong, "PONG");
 
-    let final_check: String = con_a.get("key_from_a").unwrap();
-    assert_eq!(final_check, "value_from_a");
+    // Cleanup
+    shotover_old.shutdown_and_then_consume_events(&[]).await;
```

Comment:
We should open a new connection after shutting down old shotover but before shutting down new shotover.
This will make sure that the hot reload actually occurred.